ballerina/listener_utils.bal

```diff
@@ -354,17 +354,17 @@ isolated function getHttpService(Engine gqlEngine, GraphqlServiceConfig? service
 isolated function getWebsocketService(Engine gqlEngine, readonly & __Schema schema,
                                       GraphqlServiceConfig? serviceConfig) returns UpgradeService {
     final ContextInit contextInitFunction = getContextInit(serviceConfig);
-    return isolated service object {
-        isolated resource function get .(http:Request request) returns websocket:Service|websocket:UpgradeError {
-            map<string> customHeaders = {};
-            string|http:HeaderNotFoundError subProtocol = request.getHeader(WS_SUB_PROTOCOL);
-            if subProtocol is string {
-                customHeaders = {"Sec-WebSocket-Protocol": subProtocol};
-            }
+    UpgradeService websocketUpgradeService = @websocket:ServiceConfig {
+        subProtocols: [GRAPHQL_TRANSPORT_WS]
+    } isolated service object {
+        isolated resource function get .(http:Request request) 
+        returns websocket:Service|websocket:UpgradeError|http:HeaderNotFoundError {
+            _ = check request.getHeader(WS_SUB_PROTOCOL);
             Context context = check initContext(gqlEngine, contextInitFunction, request);
-            return new WsService(gqlEngine, schema, customHeaders.cloneReadOnly(), context);
+            return new WsService(gqlEngine, schema, context);
```

Comment:
Is there any reason to not support configuring `connectionInitWaitTimeout` that is specified in the [spec](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#message-types)?

Reply:
Currently the default value from the websocket module is for `connectionInitWaitTimeout`. We can address this in a separate issue

Reply:
Issue: [#3968](https://github.com/ballerina-platform/ballerina-standard-library/issues/3968)